### ISSUE TITLE
zypper refresh not automatically importing repository signing key even with --gpg-auto-import-keys option

### ISSUE THREAD

**Issue by balasankarc** [openSUSE/zypper#144]
### OS Details
OS: openSUSE Leap 42.2 (using `opensuse:42.2` docker image)
Zypper version: 1.13.32

```bash
bash-4.3# cat /etc/os-release 
NAME="openSUSE Leap"
VERSION="42.2"
ID=opensuse
ID_LIKE="suse"
VERSION_ID="42.2"
PRETTY_NAME="openSUSE Leap 42.2"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:leap:42.2"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"

bash-4.3# zypper --version
zypper 1.13.32
```
### Environment
I have configured a repo to be used with zypper. The following shows the content of the `.repo` file:
```bash
bash-4.3# cat /etc/zypp/repos.d/gitlab_gitlab-ee.repo 
[gitlab_gitlab-ee]
name=gitlab_gitlab-ee
baseurl=https://packages.gitlab.com/gitlab/gitlab-ee/opensuse/42.2/$basearch
repo_gpgcheck=1
pkg_gpgcheck=1
enabled=1
gpgkey=https://packages.gitlab.com/gitlab/gitlab-ee/gpgkey
       https://packages.gitlab.com/gitlab/gitlab-ee/gpgkey/gitlab-gitlab-ee-3D645A26AB9FBD22.pub.gpg
autorefresh=1
type=rpm-md

[gitlab_gitlab-ee-source]
name=gitlab_gitlab-ee-source
baseurl=https://packages.gitlab.com/gitlab/gitlab-ee/opensuse/42.2/SRPMS
repo_gpgcheck=1
pkg_gpgcheck=1
enabled=1
gpgkey=https://packages.gitlab.com/gitlab/gitlab-ee/gpgkey
       https://packages.gitlab.com/gitlab/gitlab-ee/gpgkey/gitlab-gitlab-ee-3D645A26AB9FBD22.pub.gpg
autorefresh=1
type=rpm-md
```

### Problem
Even if I provide `--gpg-auto-import-keys` option to the refresh command, zypper warns about me unknown key. Theoretically, it should be simply importing and trusting the key, but that is not happening. 

```bash
bash-4.3# zypper --gpg-auto-import-keys refresh gitlab_gitlab-ee
Retrieving repository 'gitlab_gitlab-ee' metadata ----------------------------------------------------------------------------------------------------------------------------[|]
Warning: File 'repomd.xml' from repository 'gitlab_gitlab-ee' is signed with an unknown key '14219A96E15E78F4'.

    Note: Signing data enables the recipient to verify that no modifications occurred after the data
    were signed. Accepting data with no, wrong or unknown signature can lead to a corrupted system
    and in extreme cases even to a system compromise.

    Note: File 'repomd.xml' is the repositories master index file. It ensures the integrity of the
    whole repo.

    Warning: We can't verify that no one meddled with this file, so it might not be trustworthy
    anymore! You should not continue unless you know it's safe.

File 'repomd.xml' from repository 'gitlab_gitlab-ee' is signed with an unknown key '14219A96E15E78F4'. Continue? [yes/no] (no):
```

I also tried the following to see if it fixed, with no luck.
1. Run the command in non-interactive mode. Because the man page says
> 
       --gpg-auto-import-keys
           If new repository signing key is found, do not ask what to do; trust and import it automatically. This
           option causes that the new key is imported also in non-interactive mode, where it would otherwise got
           rejected.
2. Manually get the keys to my keychain using `gpg --recv-keys` and then trying zypper refresh

Any help would be appreciated as I fail to see why this happens. Thanks.

**Comment by mlandres** [comment]
Benjamins fix is merged into `zypper-1.14.8`, but it works only for missing package signing keys. Zypp now tries to retrieve them if they are listed in the repos `gpgkey=` entry.

Keys signing the repo metadata still need to be placed next to the master index file:

    repodata/repomd.xml        # master index file 
            /repomd.xml.asc    # detached signature
            /repomd.xml.key    # ascii armored key used for signing 

This should be fixed as well.

**Comment by mlandres** [comment]
This should be finally fixed in libzypp-17.6.4 by openSUSE/libzypp#133.
